apis/metal3.io/v1alpha1/baremetalhost_validation.go

```diff
@@ -309,3 +314,34 @@ func validateRebootAnnotation(rebootAnnotation string) error {
 
 	return nil
 }
+
+func validateCrossNamespaceSecretReferences(hostNamespace, hostName, fieldName string, ref *corev1.SecretReference) error {
+	if ref != nil &&
+		ref.Namespace != "" &&
+		ref.Namespace != hostNamespace {
+		return k8serrors.NewForbidden(
+			schema.GroupResource{
+				Group:    "metal3.io",
+				Resource: "baremetalhosts",
+			},
+			hostName,
+			fmt.Errorf("%s: cross-namespace Secret references are not allowed", fieldName),
```

Comment:
Should we print the offending/expected namespace here?

Reply:
I don't think we should. And what would be the expected ns in this case? The secret's or the BMH's?

Reply:
Since there is only one allowed namespace, maybe it does not make too much sense. Printing the offending namespace wouldn't hurt IMO, but I'm fine either way.